operator/controller/webhook/introute/sync.py

```diff
@@ -76,16 +79,64 @@ def get_mounts(self) -> List[Mapping]:
         return volume_mounts
 
 
-def _spring_cloud_k8s_config(parent) -> Optional[Mapping]:
+def _get_updated_props_srcs(related_configmaps, props_srcs) -> List:
+    """Replaces each props_src name with the latest ConfigMap based on creationTimestamp and prop_src base name"""
+    updated_props = []
+
+    if not props_srcs or not related_configmaps:
+        _LOGGER.debug(
+            f"Either props_srcs or configmaps were missing\n Props Sources: {props_srcs} \nConfigmaps: {related_configmaps}"
+        )
+        return updated_props
+
+    for prop_src in props_srcs:
+        matching_configmaps = {}
+        for cm_name, cm in related_configmaps.items():
+            if "name" in prop_src and prop_src["name"] in cm_name:
```

Comment:
We could potentially run into some undesirable behavior using a general substring match on the name. For example if we have two ConfigMaps named `testprops-4bc2tbtgtc` and `alt-testprops-7t5ggg8bm6`, then a `prop_src` selecting on `name = testprops` would match both the CMs, rather than only the former as would be expected.

We might want to use the `startswith()` method instead:
```
cm_name.startswith(prop_src["name"] )
```

Reply:
I think `testprops-alt-7t5ggg8bm6` would also cause some issues, so using `startswith` on its own wouldn't be enough. The hash kustomize adds to the cm name is always 10 characters in length plus 1 for the hyphen.

I think we can just strip the last 11 characters of cm_name.
```
if prop_src["name"] == cm_name[-11:]
```

Reply:
Ah yeah good point about `startswith` :facepalm:.

What about for ConfigMaps that have no hash suffix? Stripping the last 11 characters would be problematic.